addons/social_share/controllers/main.py

```diff
@@ -0,0 +1,104 @@
+import base64
+
+from odoo.http import Controller, content_disposition, request, route
+
+from .user_agents import NETWORK_TO_AGENT
+from ..utils.image_utils import scale_image
+
+class SocialShareController(Controller):
+
+    @route(['/snshare/cards/<int:campaign_id>/card.png', '/snshare/cards/<int:campaign_id>/<string:uuid>/card.png'], type='http', auth='public', sitemap=False, website=True)
```

Comment:
Can't we simplify to something like
```suggestion
    @route('/cards/<int:campaign_id>/<string:uuid>/card.png', type='http', auth='public', sitemap=False, website=True)
```
?